### PR TITLE
chore: [APPAI-1770] Setup Acceptance Tests for CLI

### PR DESCRIPTION
**Pull request by sawood14012**
Have Added Acceptance Tests for the CLI 

signed off by : msawood@redhat.com

### CODE REVIEW

File: acceptance-tests/tests/common.go [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+// CleanupGo cleans up go releated data
+func CleanupGo() {
+	err := helper.CleanupGo("data")
+	if err != nil {
+		acclog.ErrorLogger.Println(err)
+	}
+}
+
+// CleanupPypi cleans up pypi releated data
+func CleanupPypi() {
```
Comment (deepak1725) [inline acceptance-tests/tests/common.go:326]: Ideally command `crda analyse` itself should do the cleanup.. Why we have to write a separate logic for cleanup again...?

**Review by deepak1725 — CHANGES_REQUESTED** [review]

File: acceptance-tests/tests/common.go (hunk 1 above)
Comment (sawood14012) [reply acceptance-tests/tests/common.go:326]: tests work this way -> first it picks up a manifest  copies it to the data dir-> then installs the deps -> then run analyse command -> cleanup 
cleanup refers to deletion of the copied file in the data dir

Comment (deepak1725) [reply acceptance-tests/tests/common.go:326]: if this has to have for every scenario..can't it be Globally..?

Comment (sawood14012) [reply acceptance-tests/tests/common.go:326]: yup it can be written let me change

File: acceptance-tests/test.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+export AUTH_TOKEN=${THREE_SCALE_KEY};
+export CRDA_KEY=${CRDA_KEY};
+export HOST=https://f8a-analytics-preview-2445582058137.staging.gw.apicast.io
+export snyk_token=${SNYK_TOKEN}
+
+printf 'auth_token: %s\ncrda_key: %s\nhost: %s' "${AUTH_TOKEN}" "${CRDA_KEY}" "${HOST}" >> $HOME/.crda/config.yaml
+
+cat $HOME/.crda/config.yaml
+
+go get github.com/onsi/ginkgo/ginkgo
```
Comment (deepak1725) [inline acceptance-tests/test.sh:15]: why can't we install it via go mod..?
`go mod vendor`

File: acceptance-tests/tests/common.go [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+	BeARegularFile       = gomega.BeARegularFile
+	BeADirectory         = gomega.BeADirectory
+	HaveHTTPStatus       = gomega.HaveHTTPStatus
+	And                  = gomega.And
+	SatisfyAll           = gomega.SatisfyAll
+	Or                   = gomega.Or
+	SatisfyAny           = gomega.SatisfyAny
+	Not                  = gomega.Not
+	WithTransform        = gomega.WithTransform
+)
```
Comment (deepak1725) [inline acceptance-tests/tests/common.go:128]: Where exactly these variables used..? I can't see any usage at least here

File: acceptance-tests/tests/common.go [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+		acclog.ErrorLogger.Println(err)
+	}
+}
+
+// InstallNpmDeps runs npm install
+func InstallNpmDeps() {
+	cmd := exec.Command("npm", "install")
+	cmd.Dir = "data"
+	stdout, err := cmd.Output()
+	acclog.InfoLogger.Println(string(stdout))
```
Comment (deepak1725) [inline acceptance-tests/tests/common.go:175]: every log is in stdout, You can set this logging globally to `stdout` then use that in your codebase

Comment (deepak1725) [reply acceptance-tests/tests/common.go:175]: If you can, Try using the already implemented logging lib "github.com/rs/zerolog/log"

**Review by deepak1725 — CHANGES_REQUESTED** [review]

**Comment by deepak1725** [comment]
Along with that, you need to write one more variable in your config.yaml
`consent_telemetry: false`, this will help in not polluting our telemetry data

File: acceptance-tests/tests/common.go (hunk 3 above)
Comment (sawood14012) [reply acceptance-tests/tests/common.go:128]: ginkgo needs these vars its auto generated by ginkgo / if we dont want this we have to use ginkgo as `.` import which will cause linter to fail

File: acceptance-tests/test.sh (hunk 2 above)
Comment (sawood14012) [reply acceptance-tests/test.sh:15]: yup we can

File: acceptance-tests/tests/common.go (hunk 4 above)
Comment (sawood14012) [reply acceptance-tests/tests/common.go:175]: okay

**Comment by sawood14012** [comment]
> Along with that, you need to write one more variable in your config.yaml
> `consent_telemetry: false`, this will help in not polluting our telemetry data

noted.

**Comment by sawood14012** [comment]
@amitkrout done with the Changes Please Approve

**Review by amitkrout — APPROVED** [review]
Hi Heam,

I have pretty good experience of writing and executing cli test using ginkgo framework. I use my experience and did a pair review with @sawood14012. My review was focused on test structure, test execution. 

Based on my experience and above review areas i am approving this test pr.

@sawood14012 Thanks for including me in the review process.

**Review by deepak1725 — APPROVED** [review]
lgtm

**Pull request merged**